interactions.go

```diff
@@ -7,8 +7,240 @@ import (
+	Token     string                            `json:"token"`
+	Version   int                               `json:"version"`
+}
+
+// ApplicationCommandInteractionData is representing interaction data for application command.
+type ApplicationCommandInteractionData struct {
+	ID      string                                     `json:"id"`
+	Name    string                                     `json:"name"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options"`
+}
+
+// ApplicationCommandInteractionDataOption is representing an option of application's command.
+type ApplicationCommandInteractionDataOption struct {
+	Name string `json:"name"`
+	// Contains the value specified by InteractionType
+	Value   interface{}                                `json:"value,omitempty"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options,omitempty"`
+}
+
+// IntValue is utility function for casting option value to integer
+func (o ApplicationCommandInteractionDataOption) IntValue() int64 {
+	if v, ok := o.Value.(float64); ok {
+		return int64(v)
+	}
+
+	return 0
+}
+
+// UintValue is utility function for casting option value to unsigned integer
+func (o ApplicationCommandInteractionDataOption) UintValue() uint64 {
+	if v, ok := o.Value.(float64); ok {
+		return uint64(v)
+	}
+
+	return 0
+}
+
+// FloatValue is utility function for casting option value to float
+func (o ApplicationCommandInteractionDataOption) FloatValue() float64 {
+	if v, ok := o.Value.(float64); ok {
+		return float64(v)
+	}
+
+	return 0.0
+}
+
+// StringValue is utility function for casting option value to string
+func (o ApplicationCommandInteractionDataOption) StringValue() string {
+	if v, ok := o.Value.(string); ok {
+		return v
+	}
+
+	return ""
+}
+
+// BoolValue is utility function for casting option value to bool
+func (o ApplicationCommandInteractionDataOption) BoolValue() bool {
+	if v, ok := o.Value.(bool); ok {
+		return v
+	}
+
+	return false
+}
+
+// ChannelValue is utility function for casting option value to channel object.
+// s : Session object, if not nil, function additionaly fetches all channel's data
+func (o ApplicationCommandInteractionDataOption) ChannelValue(s *Session) *Channel {
+	chanID := o.StringValue()
+	if chanID == "" {
+		return nil
+	}
+
+	if s == nil {
+		return &Channel{ID: chanID}
+	}
+
+	ch, err := s.State.Channel(chanID)
+	if err != nil {
+		ch, err = s.Channel(chanID)
+		if err != nil {
+			return &Channel{ID: chanID}
+		}
+	}
+
+	return ch
+}
+
+// RoleValue is utility function for casting option value to role object.
+// s : Session object, if not nil, function additionaly fetches all role's data
+func (o ApplicationCommandInteractionDataOption) RoleValue(s *Session, gID string) *Role {
+	roleID := o.StringValue()
+	if roleID == "" {
+		return nil
+	}
+
+	if s == nil || gID == "" {
+		return &Role{ID: roleID}
+	}
+
+	r, err := s.State.Role(roleID, gID)
+	if err != nil {
```

Comment:
```
if err == nil {
  return r
}
+log (debug true) error

roles, err := s.GuildRoles(gID)
if err == nil {
    +log (debug true) error
    for _, r = range roles {
        if r.ID == roleID {
 	    return r
        }
    }
}
return &Role{ID: roleID}
```

If u can, please reduce the amount of investments

Reply:
And log error

Reply:
why?